requirements/README.rst

```diff
@@ -12,22 +12,27 @@ directly in the requirements directory.)
 
 .. _OEP-18: https://github.com/openedx/open-edx-proposals/blob/master/oeps/oep-0018-bp-python-dependencies.rst
 
-Upgrading all dependencies
-**************************
+While the ``*.in`` files are intended to be updated manually, the updating the ``*.txt`` files should only be manipulated using Makefile targets in a Linux environment (to match our build and deploy systems). Developers on Mac should prefer to use the GitHub workflows or run Make targets from inside devstack's lms-shell or another Linux environment.
 
-You can use the `upgrade-requirements Github Workflow <https://github.com/openedx/edx-platform/actions/workflows/upgrade-python-requirements.yml>`_  to make a PR that upgrades as many packages as possible.
+Workflows and Makefile targets
+******************************
 
-Upgrading just one dependency
-*****************************
+Add a dependency
+================
+
+To add a Python dependencies, specify it in the appropriate ``requirements/edx/*.in`` file, push that up to a branch, and then use the `compile-python-requirements.yml workflow <https://github.com/openedx/edx-platform/actions/workflows/compile-python-requirements.yml>`_ to run ``make compile-requirements`` against your branch. This will ensure the lockfiles are updated with any transitive dependencies and will ping you on a PR for updating your branch.
```

Comment:
```suggestion
To add a Python dependency, specify it in the appropriate ``requirements/edx/*.in`` file, push that up to a branch, and then use the `compile-python-requirements.yml workflow <https://github.com/openedx/edx-platform/actions/workflows/compile-python-requirements.yml>`_ to run ``make compile-requirements`` against your branch. This will ensure the lockfiles are updated with any transitive dependencies and will ping you on a PR for updating your branch.
```